### ISSUE TITLE
Exam performance is empty after upgrading the kolibri 0.8.0 to 0.11.0b2 

### ISSUE THREAD

**Issue by mrpau-julius** [learningequality/kolibri#4474]
### Observed behavior

When I checked the individual exam report of learners. I found out, exam performance is empty.

![exam-report](https://user-images.githubusercontent.com/38421180/47412917-c68fb700-d721-11e8-8253-170ec1323d0b.png)

![exam-report-gone](https://user-images.githubusercontent.com/38421180/47414013-c93fdb80-d724-11e8-832e-f76f8388c6c9.png)

### Errors and logs
[kolibri.log](https://github.com/learningequality/kolibri/files/2509514/kolibri.log)
[server.log](https://github.com/learningequality/kolibri/files/2509515/server.log)

### Steps to reproduce

* Install the kolibri version [0.8.0](http://pantry.learningequality.org/downloads/kolibri/v0.8.0/) 
* Sign-in as admin and create a class, at least one or two learners and one exam.
* Then sign-in as learner and answered completely the exam.
* After that is done, Upgrade the kolibri to [0.11.0b2](https://github.com/learningequality/kolibri/releases/tag/v0.11.0-beta2)
* Sign-in as admin go to Coach > Exams page
* Then check the individual learners exam performance

### Context

Kolibri version: upgrade 0.8.0 to 0.11.0b2
Operating system: windows 10
Browser: firefox 62.0.3 (64-bit)

**Comment by jonboiser** [comment]
The Exam Log looks like it's fine (since the gold star appears on the exam report).

The logs are showing errors when trying to access `/api/content/contentnode/node_assessments/`

getting this error:

```
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\backends\utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "C:\Python34\lib\site-packages\kolibri\dist\django\db\backends\sqlite3\base.py", line 328, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: too many SQL variables
```

**Comment by rtibbles** [comment]
Was that the complete URL, or are there get params on the request also?

**Comment by jonboiser** [comment]
That's from the `kolibri.log` file. The full line is

`ERROR 2018-10-23 03:19:14,351 exception Internal Server Error: /api/content/contentnode/node_assessments/`

But I'm not sure how this endpoint got hit. It only looks like it should be used  during exam creation.

**Comment by rtibbles** [comment]
Yeah, that was what I thought too. Can you check that `fetchNodeAssessments` is not being called anywhere it shouldn't be?

**Comment by jonboiser** [comment]
I can't reproduce this on Mac. I'll try this again on a windows computer.

**Comment by radinamatic** [comment]
Not able to reproduce this on Ubuntu either. 
Same scenario, Kolibri updated from 0.8 to 0.11b2, performance & results of the exam taken by a learner while on Kolibri 0.8 are clearly viewable after the upgrade:

![ubuntu 16 04 3 0 8 with users content running - oracle vm virtualbox_090](https://user-images.githubusercontent.com/1457929/47467862-96ffae00-d7f8-11e8-9dd2-4dc0ee923470.png)

![ubuntu 16 04 3 0 8 with users content running - oracle vm virtualbox_091](https://user-images.githubusercontent.com/1457929/47467869-9d8e2580-d7f8-11e8-80d6-5d5aeebd9a71.png)

**Comment by jonboiser** [comment]
@mrpau-julius 	Could you check the data (and any logs) from this API response

`http://localhost:8080/api/logger/examattemptlog/?exam=EXAM_ID&user=USER_ID`

If the Exam Report URL on your browser is something like

http://localhost:8080/coach/#/108687d5292e2e44d40e331505ad9883/exams/149cc268bbf2f54b2b5a750df314a3d5/users/4090238f75ffe99237af05ae08acd3c0/0/0

Then `EXAM_ID` is the long string after `.../exams/` and `USER_ID` is the long string after `.../users/`. Replace these in 

http://localhost:8080/api/logger/examattemptlog/?exam=EXAM_ID&user=USER_ID

so we can see if the correct exam progress logs are coming back.

**Comment by jonboiser** [comment]
@mrpau-julius 	Could you also send the response from

http://localhost:8080/api/exams/exam/149cc268bbf2f54b2b5a750df314a3d5/

where you replace that last string with your EXAM_ID.

**Comment by mrpau-julius** [comment]
@jonboiser Sure, I will check that today.

**Comment by mrpau-julius** [comment]
Hi @jonboiser here are the result. The field value are blank.

![api1](https://user-images.githubusercontent.com/38421180/47480916-d2917c80-d7e6-11e8-8b09-928b74bad53b.png)

![api2](https://user-images.githubusercontent.com/38421180/47480936-df15d500-d7e6-11e8-86e3-1edf5070973e.png)

**Comment by jonboiser** [comment]
@mrpau-julius Could you try it again, but log in as the same user. Click the "Log in" button at the top right of the Django Rest Framework UI, sign in as the same user in your first post (or as a superuser), then check the same API responses.

The Exam Attempt Log List does look odd because it returned an empty array, but I'm not sure if that's because you are not logged in.

**Comment by mrpau-julius** [comment]
Hi @jonboiser sure, this are the result.
This is the filter from `Exam Attempt Log List`[Exam Attempt Log List.txt](https://github.com/learningequality/kolibri/files/2517764/Exam.Attempt.Log.List.txt), on the bottom the field are blank.
![exam attempt log list](https://user-images.githubusercontent.com/38421180/47545624-1d23ff00-d8a2-11e8-9367-d6db57609bc1.png)

And another filter from `Exam Viewset Instance`[Exam Viewset Instance.txt](https://github.com/learningequality/kolibri/files/2517774/Exam.Viewset.Instance.txt), on the bottom the field have a value.
![exam viewset instance](https://user-images.githubusercontent.com/38421180/47545715-928fcf80-d8a2-11e8-96e2-cdd152b3dc43.png)

**Comment by radinamatic** [comment]
Not having much luck with this... After replicating the scenario with the provided 0.8 `.kolibri` folder, I'm unable to restart the server when I upgrade to 0.11:

![win10 0 8 running - oracle vm virtualbox_136](https://user-images.githubusercontent.com/1457929/47818181-ce6ddd80-dd57-11e8-838b-ec89d8b1fe67.png)

[debug.log](https://github.com/learningequality/kolibri/files/2536370/debug.log)
[kolibri.log](https://github.com/learningequality/kolibri/files/2536371/kolibri.log)

**Comment by radinamatic** [comment]
After several failed attempts I gave up on trying to upgrade the 0.8 installation that was running with the sample database & content handed over by @mrpau-dev. Had issues displaying content of the exam report in question even on 0.8, and was unable to build other exams from the content that was there, supposedly. Only after I downloaded more content, it was possible to build another exam. I suspect content in that home folder got corrupted, at least, so not necessarily an exam issue either... 

I tried to upgrade that 0.8 instance to 0.11, but server wouldn't start (screenshot from the post above, when I run from the command line).

At that point I gave up and replicated the whole upgrade scenario from scratch on a fresh VM:
1. Install 0.8, download some content, create users, build an exam with 50 questions, take the exam as a learner, review and check the exam report as coach/admin - no issues.
2. Upgrade to 0.11b2, open the exam taken on 0.8, and review the exam report - displayed without issues too.

![win10 0 8 running - oracle vm virtualbox_148](https://user-images.githubusercontent.com/1457929/47889491-8ed3ee00-de4a-11e8-8358-18204619b141.png)


@jonboiser I was officially unable to replicate the error in an useful way to approach fixing it... :disappointed: 
Let me know if you think there's something else I should try!
